File: projects/06/ArrayProcessing.java
/* ArrayProcessing.java prints an array and fills an array from user input
 * by: Bryce Allen
 * cs 214 
 * Calvin College
 * 3/26/18
 */

import java.util.Scanner;


public class ArrayProcessing {
    public static void main(String[] args)
    {
	Scanner keyboard = new Scanner(System.in);
	System.out.println("Enter number of values for the array: ");
	int size = keyboard.nextInt();
	double[] list = new double[size];
	System.out.println("Enter the values: ");
	enterValues(list);//enter value subprogram
	printValues(list);//print value subprogram
    }

    public static void enterValues(double[] anArray)
    {
	Scanner scan = new Scanner(System.in);
	for(int i = 0; i < anArray.length; i++)
	    {
		anArray[i] = scan.nextDouble();
	    }
    }

    public static void printValues(double[] anArray)
    {
	for(int i = 0; i < anArray.length; i++)
	    {
		System.out.print(" " + anArray[i] + " ");
	    }
    }
}

File: projects/06/array_processing.rb
## array_processing.rb processes (reads and write values) to an array.
## By: Bryce Allen
## cs 214
## Calvin College
## 3/27/18
##########################################################################

def writeValues(anArray)
  for i in 0...anArray.size
      anArray[i] = gets.chomp.to_f
    end
end

def readValues(anArray)
  for i in 0...anArray.size
    puts anArray[i]
  end
end

  
def main
  puts "Enter number of values for the array: "
  size = gets.chomp.to_f
  arr = Array.new(size)
  puts "Enter the values: "
  writeValues(arr);
  puts "values are: "
  readValues(arr);
end

main
